components/partners/partnerrepository/src/main/java/org/mozilla/lite/partner/Repository.java

```diff
@@ -50,17 +52,23 @@ public void loadMore() {
         nextSubscription();
     }
 
-    // Not used yet and not tested yet.
     public void reloadData() {
         itemPojoList = new ArrayList<>();
         cacheIsDirty = true;
+        currentPage = firstPage;
```

Comment:
can you call reset and then ` nextSubscription();` ?

Reply:
For this case, it should not set onDataChangedListener as null in reset()